bgpd/bgp_fsm.c

```diff
@@ -722,9 +723,13 @@ bgp_start (struct peer *peer)
   /*
    * Don't connect if we haven't received interface_up message from zebra
    */
+  if (sockunion_family (&peer->su) == AF_INET)
+    afi = AFI_IP;
+  else if (sockunion_family (&peer->su) == AF_INET6)
+    afi = AFI_IP6;
+
```

Comment:
what if sockunion_family is not AF_INET or AF_INET6
Please add this case and don't run the check for other address families.

Reply:
done

Reply:
In this case you'll stop receiving sessions from non IPv4 or IPv6.
I'd better to check if we have IPv4 or IPv6 session, and if true, make your check, if not, just allow it keep going.
something like
```
afi_t afi = AFI_UNSPEC;
if (sockunion_family (&peer->su) == AF_INET) 
    afi = AFI_IP; 
  else if (sockunion_family (&peer->su) == AF_INET6) 
    afi = AFI_IP6;
if afi != AFI_UNSPEC && ...
```

Reply:
afi has another value AFI_MAX defined as well. so planning to check explicitly for ipv4 and ipv6 instead of the not condition